subsys/dfu/src/dfu_target.c

```diff
@@ -58,14 +55,16 @@ int dfu_target_init(int img_type, size_t file_size, dfu_target_callback_t cb)
 {
 	const struct dfu_target *new_target = NULL;
 
-	if (IS_ENABLED(CONFIG_DFU_TARGET_MCUBOOT) &&
-	    img_type == DFU_TARGET_IMAGE_TYPE_MCUBOOT) {
+#ifdef CONFIG_DFU_TARGET_MCUBOOT
+	if (img_type == DFU_TARGET_IMAGE_TYPE_MCUBOOT) {
 		new_target = &dfu_target_mcuboot;
-	} else if (IS_ENABLED(CONFIG_DFU_TARGET_MODEM) &&
-		   img_type == DFU_TARGET_IMAGE_TYPE_MODEM_DELTA) {
+	}
+#endif
+#ifdef CONFIG_DFU_TARGET_MODEM
+	if (img_type == DFU_TARGET_IMAGE_TYPE_MODEM_DELTA) {
```

Comment:
Will it? I'm not sure that it will.

I agree that the original config does a logic precedence and it's not preserved in the change but it will not overwrite `new_target` since `img_type` can't be both `DFU_TARGET_IMAGE_TYPE_MCUBOOT` and `DFU_TARGET_IMAGE_TYPE_MODEM_DELTA` at the same time when this function is called.

Reply:
The http_application_update sample in master set below flags:
CONFIG_DFU_TARGET=y
CONFIG_DFU_TARGET_MCUBOOT=y
CONFIG_DFU_TARGET_MODEM=y
My understanding is that the actual type is decided from the meta data of the downloaded image, which means both are possible. If it is TARGET_MCUBOOT, above code might change it to TARGET_MODEM, while original code would not.

Reply:
Okei cool. Thanks for looking into it :)